cmd/api/main.go

```diff
@@ -63,14 +66,56 @@ func main() {
 	if err != nil {
 		log.Printf("Could not start opentelemetry: %s", err)
 	}
+
 	db, err := database.NewDatabase()
 	if err != nil {
 		log.Fatalf("Could not connect to database: %s", err)
 	}
 	controller := routers.Controller{Database: db}
-	server := NewServer(getKubernetesClient(), controller)
+
+	cacheExpirations := getCacheExpirations()
+	cache := cache.New()
+
+	server := NewServer(getKubernetesClient(), controller, cache, cacheExpirations)
 	err = server.router.RunTLS("localhost:8081", "/etc/kubearchive/ssl/tls.crt", "/etc/kubearchive/ssl/tls.key")
 	if err != nil {
 		log.Printf("Could not run server on localhost: %s", err)
 	}
 }
+
+const (
+	cacheExpirationAuthorizedKey   = "CACHE_EXPIRATION_AUTHORIZED"
+	cacheExpirationUnauthorizedKey = "CACHE_EXPIRATION_UNAUTHORIZED"
+)
+
+type cacheExpirations struct {
+	Authorized   time.Duration
+	Unauthorized time.Duration
+}
+
+func getCacheExpirations() *cacheExpirations {
+	expirationAuthorizedString := os.Getenv(cacheExpirationAuthorizedKey)
+	if expirationAuthorizedString == "" {
+		log.Fatalf("The environment variable '%s' needs to be set.", cacheExpirationAuthorizedKey)
+	}
+
+	expirationAuthorized, err := time.ParseDuration(expirationAuthorizedString)
+	if err != nil {
+		log.Fatalf("'%s': '%s' could not be parsed into a duration: %s", cacheExpirationAuthorizedKey, expirationAuthorizedString, err)
+	}
+
+	expirationUnauthorizedString := os.Getenv(cacheExpirationUnauthorizedKey)
+	if expirationUnauthorizedString == "" {
+		log.Fatalf("The environment variable '%s' needs to be set.", cacheExpirationAuthorizedKey)
```

Comment:
There is a typo here:
```suggestion
		log.Fatalf("The environment variable '%s' needs to be set.", cacheExpirationUnauthorizedKey)
```

Reply:
Good catch!